### PR TITLE
Week07 signup page

### LINKED ISSUE #167: Sign Up Page

### PR DESCRIPTION
**Pull request by badgerbadgers**
Ok my largest PR yet. This one's a doozy, contains a new component, features, redesign new font and elements! My task from last week where a user can register with email and password involved more changes tweaks and research than I originally thought.
 
First some new UI changes the SignIn component now has a form to that allows the user to sign in with an email and password.
Screen Shot 2022-09-12 at 10 18 04 PM

There is a new SignUp component where the user can sign up for a new account with an email and password. There is currently no strict validation here, you can any any email and weak password, will add secure rules possibly a new todo for the board. Once the user clicks the sign up button that user will be authenticated and stored on firebase.
Screen Shot 2022-09-12 at 10 26 58 PM

Firebase authentication with user created email
Screen Shot 2022-09-12 at 10 27 58 PM

Finally the user signed in with email and password can send messages. Currently the displayName that will appear in the text message is the user's email address.
Screen Shot 2022-09-12 at 10 30 43 PM

Still working on tweaks to these features but this PR closes #167 and #168.

### CODE REVIEW

File: src/components/MiddleChatWindow.js [hunk 1]
```diff
@@ -91,7 +90,7 @@ function MiddleChatWindow({ currentUser }) {
               <img
                 className='user-image'
                 // Conditional statement for profile image
-                src={message ? message.profilePicUrl : ""}
+                src={message ? message.profilePicUrl : UserLogo}
```
Comment (viniciuspietscher) [inline src/components/MiddleChatWindow.js:93]:
```suggestion
                src={message.profilePicUrl ? message.profilePicUrl : UserLogo}
```

**Comment by viniciuspietscher** [comment]
https://user-images.githubusercontent.com/8095975/189966872-dd68b40d-928b-48d7-b155-b45a83f0f6bc.mp4

I think I was able to fix it!
Will have to fix a few other things, but will push the changes soon.

**Review by asaiapalacios — COMMENTED** [review]
Hi. Kudos for all the work put in. Is there a way to center the Sparrow logo of the sign-in page?

**Review by asaiapalacios — COMMENTED** [review]
I would recommend, for the next iteration of the desktop version, to decrease the width of the input fields and the buttons, so as to not appear to long in length.
desktop-version

Resembling more so this version:
desktop-version2

**Review by asaiapalacios — COMMENTED** [review]
Also, great job on the layout for the email sign up/sign in & the google sign in (high-five)!

**Comment by badgerbadgers** [comment]
Just seeing alot of the comments didn't get notification. @asaiapalacios Yes feel free to play around with layout. I wanted half of the screen to display the logo while the other half has the form would look cool to me. Nothing is set and I just wanted something that looks ok and not style the forms too long. Thank you!

**Comment by asaiapalacios** [comment]
Yes, will do - play around with the layout. Thanks

**Pull request merged**